### PR TITLE
$RUN_IN_GIT_ROOT for git-ship

### PR DESCRIPTION
**Pull request by allewun**
@kevgo @charlierudolph 

Seeing as ~~`git-extract`, `git-hack`, and~~ `git-ship` perform a sync, and the actual `git-sync` command requires `RUN_IN_GIT_ROOT=true`, I think these need to be set.

This fixes a bug I encountered with `git-ship`:
```bash
master$ git hack feature
feature$ mkdir new-directory
feature$ cd new-directory
feature$ touch new-file
feature$ git add -A && git commit -m "added new file to a new directory"
feature$ git ship

[feature] git fetch --prune

[feature] git checkout master
Switched to branch 'master'
Your branch is up-to-date with 'origin/master'.
fatal: Unable to read current working directory: No such file or directory
fatal: Unable to read current working directory: No such file or directory
fatal: Unable to read current working directory: No such file or directory
fatal: Unable to read current working directory: No such file or directory
fatal: Unable to read current working directory: No such file or directory
fatal: Unable to read current working directory: No such file or directory

[] git push -u origin master
fatal: Unable to read current working directory: No such file or directory

To abort, run "git ship --abort".
To continue after you have resolved the conflicts, run "git ship --continue". 

$
```

**EDIT:** Removing support for `git-extract` and  `git-hack` from this PR. Future of `git-extract` unclear (#601).

### CODE REVIEW

**Comment by charlierudolph** [comment]
Need tests for this. Probably just in shared. Technically any command that changes branches will need it. This PR just addressing these three is good though.

**Comment by allewun** [comment]
Yeah, I'm going to move the ones in git-sync/ out into shared/.

File: features/shared/folder_does_not_exist_on_main_branch/conflict.feature [hunk 1]
```diff
@@ -13,11 +13,11 @@ Feature: git sync: syncing inside a folder that doesn't exist on the main branch
       | other-feature   | local and remote | other feature commit       | file2            |                 |
     And I am on the "current-feature" branch
     And I have an uncommitted file
-    When I run `git sync --all` in the "new_folder" folder
 
 
   @finishes-with-non-empty-stash
-  Scenario: result
+  Scenario: git-sync
+    When I run `git sync --all` in the "new_folder" folder
```
Comment (kevgo) [inline features/shared/folder_does_not_exist_on_main_branch/conflict.feature]: Why do you run `git sync --all` here, and not just `git sync`?

Comment (kevgo) [inline features/shared/folder_does_not_exist_on_main_branch/conflict.feature]: Why did you move this step from the background into each of the scenarios? This is now less dry.

Comment (allewun) [inline features/shared/folder_does_not_exist_on_main_branch/conflict.feature]: I was preparing to add tests for git-ship, hack, and extract to the same feature. Didn't get to them yet.

File: features/shared/folder_does_not_exist_on_main_branch/no_conflict.feature [hunk 2]
```diff
@@ -13,11 +13,21 @@ Feature: git sync: syncing inside a folder that doesn't exist on the main branch
...
       | other-feature   | local and remote | other feature commit | file2            |
     And I am on the "current-feature" branch
-    And I have an uncommitted file
-    When I run `git sync --all` in the "new_folder" folder
 
 
-  Scenario: result
+  Scenario: git-ship
+    When I run `git ship -m 'feature done'` in the "new_folder" folder
+    Then I am on the "main" branch
```
Comment (charlierudolph) [inline features/shared/folder_does_not_exist_on_main_branch/no_conflict.feature]: Please also verify the commands that are run

**Comment by charlierudolph** [comment]
Closing in favor of #711. Reopen if you disagree.
